core/src/main/java/com/linecorp/armeria/client/cookie/CookieJar.java

```diff
@@ -27,12 +27,29 @@
 public interface CookieJar {
 
     /**
-     * Returns the {@link Cookies} for the specified {@link URI}.
+     * Returns the unexpired {@link Cookies} for the specified {@link URI}.
      */
     Cookies get(URI uri);
 
     /**
      * Stores the specified {@link Cookies} for the {@link URI}.
      */
-    void set(URI uri, Cookies cookies);
+    void set(URI uri, Iterable<? extends Cookie> cookies);
+
+    /**
+     * Stores the specified {@link Cookies} for the {@link URI} given the creation time.
+     */
+    void set(URI uri, Iterable<? extends Cookie> cookies, long createdTimeMillis);
+
+    /**
+     * Determines if a cookie is expired. Throws an {@link IllegalArgumentException} if the cookie is not
+     * found in the jar.
```

Comment:
How about defining an enum like `CookieState` rather than raising an exception?

```java
public enum CookeState {
  EXISTENT,
  EXPIRED,
  NON_EXISTENT
}
```

Then we could rename `isExpired()` to `state()`?